torch/csrc/jit/codegen/cuda/transform_iter.cpp

```diff
@@ -67,18 +67,42 @@ void ReplayTransformations::handle(Merge* m) {
   // we're going to replay the merge on
   auto it_outer = id_map_.find(id_outer);
   auto it_inner = id_map_.find(id_inner);
-  if (it_outer == id_map_.end() || it_inner == id_map_.end()) {
-    if (check_all_ops_run_) {
-      TORCH_INTERNAL_ASSERT(
-          false, "Transform traversal failed, dependencies not met.");
-    } else {
-      return;
+
+  bool outer_found = it_outer != id_map_.end();
+  bool outer_bcast = id_outer->isBroadcast();
+  bool inner_found = it_inner != id_map_.end();
+  bool inner_bcast = id_inner->isBroadcast();
+
+  // If either are not found
+  if ((!outer_found || !inner_found)) {
+    // If both aren't found, it's a failure
+    // If outer is found && inner is bcast it is not a failure
+    // If inner is found && outer is bcast it is not a failure
+    if (!(outer_found || inner_found) || (outer_found && !inner_bcast) ||
+        (inner_found && !outer_bcast)) {
+      if (error_on_failure_) {
+        TORCH_INTERNAL_ASSERT(
+            false, "Transform traversal failed, dependencies not met.");
+      } else {
+        return;
+      }
     }
   }
 
+  // If we merge a broadcast dim with a non-broadcast dim, just remap the output
+  // to the non-broadcast dim.
+  if (inner_found && !outer_found && outer_bcast) {
+    id_map_[m->out()] = it_inner->second;
+    return;
+  }
+  if (outer_found && !inner_found && inner_bcast) {
+    id_map_[m->out()] = it_outer->second;
+    return;
+  }
+
   // Grab the IDs we're going to replay this merge on
-  auto id_outer_mapped = (*it_outer).second;
-  auto id_inner_mapped = (*it_inner).second;
+  auto id_outer_mapped = it_outer->second;
```

Comment:
const?

Reply:
oh cool, I thought it wouldn't work in `leaf_ids_.find(...)` but does fine. Thanks.